tool/lib/commands/pub_get.dart

```diff
@@ -67,6 +68,10 @@ class PubGetCommand extends Command {
         ['pub', command],
         workingDirectory: p.packagePath,
       );
+      // For me (dantup), these processes never complete unless we are
+      // consuming stdout too, so run it through log.trace.
+      process.stdout.transform(utf8.decoder).listen(log.trace);
```

Comment:
should we just use the ProcessManager instead like some of the other commands? I think the ProcessManager class automatically manages forwarding the stdout and stderr streams

Reply:
ProcessManager didn't expose stderr, so I had to refactor it a little. I moved the analyzer command over to use it too.

I bumped the language version to 3.0 in tool/ so I could use a record for the return type (the SDK `ProcessResult` has `dynamic` fields!). If this is a problem, lmk and I can switch it to a class.